### ISSUE TITLE
Colors broken

### ISSUE THREAD

**Issue by bluehawk** [auraphp/Aura.Cli#29]
A recent change broke the color rendering in Aura for us. 

Our setup is that we are using PuTTY to SSH into a Gentoo server, not sure if that's relevant. The problem is present in both PHP version `PHP 5.3.28-pl1-gentoo (cli) (built: Mar 10 2014 18:34:15)` and `PHP 5.4.23-pl0-gentoo (cli) (built: Mar 10 2014 18:17:14)`

I did a bisect, and the offending commit seems to be:

```
ddcb86be93c9b8ed6991fe98a9d5733970a58399 is the first bad commit
commit ddcb86be93c9b8ed6991fe98a9d5733970a58399
Author: Paul M. Jones <pmjones88@gmail.com>
Date:   Tue Apr 15 20:42:39 2014 -0500

    soothe the scrutinizer
```

Here are some screenshots that show this:

(./minion is an executable script we created to run cli commands using Aura)

![image](https://cloud.githubusercontent.com/assets/139870/2766646/f6d5effe-ca30-11e3-9792-077c95bbd68a.png)

![image](https://cloud.githubusercontent.com/assets/139870/2766653/033b3542-ca31-11e3-89a7-3317b976f144.png)

![image](https://cloud.githubusercontent.com/assets/139870/2766658/16c1883c-ca31-11e3-9dbc-5b39a9dd47a0.png)

![image](https://cloud.githubusercontent.com/assets/139870/2766664/2053ab6e-ca31-11e3-98df-cabe0c70b836.png)

It seems that [casting the file descriptor to an int](https://github.com/auraphp/Aura.Cli/commit/ddcb86be93c9b8ed6991fe98a9d5733970a58399#diff-4d3c3a9c9fc594721153f8a1acfeabf1R137) is somehow making Aura think it isn't a TTY anymore. I'm not really sure why that happens, since the php docs say that you should pass in an int. 

What was the context for casting it to an int in order to "soothe the scrutinizer"? If casting to an int is necessary for scrutinizer to pass, the following line could be a solution:

```
 $this->posix = posix_isatty($this->resource) OR posix_isatty((int) $this->resource);
```

**Comment by harikt** [comment]
You are absolutely right. https://github.com/auraphp/Aura.Cli/commit/ddcb86be93c9b8ed6991fe98a9d5733970a58399 is broken for converting a stream to integer .

Tested code is

``` php
<?php
$resource = fopen('php://stdout', 'w+');
var_dump($resource);
var_dump((int) $resource);
var_dump(posix_isatty($resource));
var_dump(posix_isatty((int) $resource));
fclose($resource);
```

@pmjones is on travel and may be away for the week. If you like to get credited on the contributors list can send a PR removing the type cast to int . He will check it and merge if found good, else will find a different way if there is some other way to resolve the situation.

Looking forward.

Thanks

**Comment by pmjones** [comment]
Yes please send PR. Short version is that Scritinizer thought it was a bug.

**Comment by bluehawk** [comment]
Heh, ignore the first couple commits listed above, my git user name and email settings were wrong.

I've created a pull request #30

**Comment by pmjones** [comment]
Fixed by #30 -- thanks, and sorry for the hassle.
